### DISCUSSION TITLE
How to change global styles conditionally with CSS Modules?

### DISCUSSION THREAD

**Discussion by jensmeindertsma** [vercel/next.js#12967] [Help] [answered]
I have a `Navigation` component that needs to lock the body scroll with `overflow: hidden` when the navigation menu is open. I'm converting my project from `styled-components` to CSS Modules and I've come across this piece. How can I apply styles to the body conditionally from my Navigation component that has a style file `Navigation.module.css`?

I've read that global styles that apply to the body or `#__next` are only possible by importing a stylesheet into `_app.js`. But how would I manipulate these styles from a child component?

I've read through pretty much all the content there is about CSS Modules and Next.js. @Timer mentioned in an issue that Vercel is refactoring their styling to use the built in CSS support too. They are a large company so they must have solved this problem _somewhere_. I'm looking for some advice from people who have solved this issue.

**Comment by Ra-C86** [comment]
I'm looking for the same solution have not found one yet

**Reply by brunocrosier** [reply]
Could you do something like:

```javascript

<Navigation className={shouldLock ?  foo  : bar } />

```

where `foo` and `bar` are different CSS modules, and `shouldLock` is based on whether the navigation is open ?

Alternatively, what about adding a `<style jsx global>` component conditionally ?

**Reply by jensmeindertsma** [reply]
Your approach would work for changing styles dynamically on the Navigation element, but not to change styles on the `<body>`. I believe `styled-jsx` is the best solution for this, especially since it's already bundled with Next.js

**Comment by jensmeindertsma** [comment, marked as answer]
> Could you do something like:
> 
> ```js
> <Navigation className={shouldLock ?  foo  : bar } />
> ```
> 
> where `foo` and `bar` are different CSS modules, and `shouldLock` is based on whether the navigation is open ?
> 
> Alternatively, what about adding a `<style jsx global>` component conditionally ?



Closing because question is answered

**Reply by brunocrosier** [reply]
not sure why you didn't select my solutin as the answer 🤔

**Reply by jensmeindertsma** [reply]
@brunocrosier I can't select your solution as the answer since your solution is a reply to someone else's answer, so I figured this was the best way to point to your solution as the answer for anyone viewing this question later
